pychunkedgraph/graph/edits.py

```diff
@@ -94,12 +94,21 @@ def merge_preprocess(
     Get subgraph within the bounding box
     Add fake edge if there are no inactive edges between two components.
     """
-    edges_roots = cg.get_roots(
-        subgraph_edges.ravel(), assert_roots=True, time_stamp=parent_ts
-    ).reshape(-1, 2)
-    # active : edges belong to same root
-    active_mask = edges_roots[:, 0] == edges_roots[:, 1]
-    active, inactive = subgraph_edges[active_mask], subgraph_edges[~active_mask]
+    edge_layers = cg.get_cross_chunk_edges_layer(subgraph_edges)
+    active_edges = [types.empty_2d]
+    inactive_edges = [types.empty_2d]
+    for layer in np.unique(edge_layers):
+        layer_edges = subgraph_edges[edge_layers == layer]
+        edges_parents = cg.get_roots(
+            layer_edges.ravel(), time_stamp=parent_ts, stop_layer=layer+1
```

Comment:
```stop_layer=layer+1```
why +1?

Please add that answer to the code as documentation as well for future reference

Reply:
Because the parent is at the next layer of cross chunk edge, if a cross chunk edge connect 2 IDs at layer 5 for example, parent is at 6.
Same for in chunk edges, cross chunk edge layer is 1 so parent is L2